solr/core/src/java/org/apache/solr/servlet/PriorityBasedRateLimiter.java

```diff
@@ -27,11 +30,22 @@ public class PriorityBasedRateLimiter extends RequestRateLimiter {
 
   private final long waitTimeoutInNanos;
 
-  public PriorityBasedRateLimiter(RateLimiterConfig rateLimiterConfig) {
+  private SolrMetricsContext solrMetricsContext;
+
+  private final Timer foregroundRequestDelay;
+  private final Timer backgroundRequestDelay;
+
+  public PriorityBasedRateLimiter(
+      RateLimiterConfig rateLimiterConfig, SolrMetricsContext solrMetricsContext) {
     super(rateLimiterConfig);
     this.numRequestsAllowed = new Semaphore(rateLimiterConfig.allowedRequests, true);
     this.totalAllowedRequests = rateLimiterConfig.allowedRequests;
     this.waitTimeoutInNanos = rateLimiterConfig.waitForSlotAcquisition * 1000000l;
+    this.initializeMetrics(solrMetricsContext, null);
+    this.foregroundRequestDelay =
+        solrMetricsContext.timer("foregroundRequestDelay", "PriorityBasedRateLimiter");
+    this.backgroundRequestDelay =
+        solrMetricsContext.timer("backgroundRequestDelay", "PriorityBasedRateLimiter");
```

Comment:
gotcha: this is still using the _parent_ metricsContext passed in! We should instead use `this.solrMetricsContext`, created in `initializeMetrics()`.